### PR TITLE
multi_interval

### PR DESCRIPTION
**Pull request by padrone1225**

### CODE REVIEW

**Review by padrone1225 — COMMENTED** [review]
Have one issue

File: src/components/body/index.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        .then((response) => {
+          let list: any[] = [];
+          response.data.map((data: any, index: number) =>
+            list.push(data.author_association)
+          );
+          list = list.splice(0, 5);
+          setData(list);
+        })
+        .catch((error) => {})
+        .finally(() => {});
```
Comment (stevenLee818) [inline src/components/body/index.tsx:23]: Why do we need this?

File: src/components/body/index.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from "react";
+import { InputCode } from "../inputCode";
+import { ShowName } from "../showNames";
+import axios from "axios";
+import "./index.css";
+
+export const Body = () => {
+  const [view, setView] = useState(false);
```
Comment (stevenLee818) [inline src/components/body/index.tsx:8]: `visible` might be better name.

File: src/components/body/index.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  const [timeFlag, setTimeflag] = useState(false);
+  const [data, setData] = useState<any[]>([]);
```
Comment (stevenLee818) [inline src/components/body/index.tsx:10]: `any` is strongly discouraged in TypeScript

File: src/components/body/index.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  useEffect(() => {
+    if (view && !timeFlag) {
+      axios("https://api.github.com/repos/elixir-lang/elixir/issues")
+        .then((response) => {
+          let list: any[] = [];
```
Comment (stevenLee818) [inline src/components/body/index.tsx:15]: same here

File: src/components/body/index.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+          response.data.map((data: any, index: number) =>
```
Comment (stevenLee818) [inline src/components/body/index.tsx:16]: same here

Comment (stevenLee818) [inline src/components/body/index.tsx:16]: Why don't you use the return value of Array.map?

File: src/components/body/index.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        setView(false);
+      }, 15000);
+    } else if (!view) {
+      setTimeflag(false);
+      setData([]);
+      return clearTimeout(0);
+    }
+  }, [timeFlag, view]);
+  return (
+    <div id="body" className="body container">
```
Comment (stevenLee818) [inline src/components/body/index.tsx:35]: do we need id here?

File: src/components/inputCode.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
+import { ChangeEvent, useEffect, useRef, useState } from "react";
+
+export const InputCode = ({ setCode }: { setCode: Function }) => {
```
Comment (stevenLee818) [inline src/components/inputCode.tsx:3]: do proper typing rather than `Function`

File: src/components/inputCode.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    //   //     setTimeflag(true);
+    //   //     setCountTime(countTime + 1);
+    //   //     if (countTime === 3) {
+    //   //       setPassword("");
+    //   //       setCountTime(0);
+    //   //     }
+    //   //   }, 1000);
+    //   setIntervalId(tiemww);
+    //   //   return () => clearInterval(tiemww);
+    // }
```
Comment (stevenLee818) [inline src/components/inputCode.tsx:59]: ✂️

File: src/components/inputCode.tsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  }, [password]);
+  return (
+    <input
+      type="text"
+      className="codeinput"
+      onKeyDown={(e) => handleKeydown(e)}
+      onChange={handleChange}
```
Comment (stevenLee818) [inline src/components/inputCode.tsx:66]: handlePasswordChange

File: src/components/header/index.tsx [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import "./index.css";
+
+export const Header = () => {
+  return (
+    <div id="header" className="header container">
```
Comment (stevenLee818) [inline src/components/header/index.tsx:5]: id is needed?

**Review by stevenLee818 — COMMENTED** [review]
There are many typing issues. Please fix it. On the other hand, please check https://www.npmjs.com/package/react-konami-code and its source

**Pull request merged**